### DISCUSSION TITLE
How does setting manual control work?

### DISCUSSION THREAD

**Discussion by tamueller** [basnijholt/adaptive-lighting#257] [Q&A] [answered]
I have an button/automation that set's a certain lamp to 10% brightness. If i use the button while the lamp is on, it stays at 10%, according the manual takeover logic.
However, when I use the button while the lamp is off, the lamp turns to it's default brightness after a few seconds (as expected).

I now tried setting the light to manual control like this, but this doesn't have any effect:

  - service: adaptive_lighting.set_manual_control
    data:
      entity_id: light.bad_decke
      manual_control: true

Also in the manual there is a field "entity id" which I don't really understand. "The entity_id of the switch in which to (un)mark the light as being "manually controlled""
Is that necessary, and if so, what do I need to put there if the automation is triggered by multiple switches?

**Comment by tamueller** [comment, marked as answer]
Since noone has answered yet: the entity_id is the id of the adaptive light instance, e.g.            
   entity_id: switch.adaptive_lighting_instance_1

**Reply by MattDemers** [reply]
Thanks for this; I've been trying to figure out an automation I've been working with. Maybe this needs to be added to documentation.

**Comment by th3w1zard1** [comment]
The documentation for `adaptive-lighting.set_manual_control()` is clearly documented in the README file:

`adaptive_lighting.set_manual_control` can mark (or unmark) whether a light is "manually controlled", meaning that when a light has `manual_control`, the light is not adapted.

| Service data attribute | Optional | Description                                                                                         |
| ---------------------- | -------- | --------------------------------------------------------------------------------------------------- |
| `entity_id`            | no       | The `entity_id` of the switch in which to (un)mark the light as being "manually controlled".        |
| `lights`               | yes      | entity_id(s) of lights, if not specified, all lights in the switch are selected.                    |
| `manual_control`       | yes      | Whether to add ('true') or remove ('false') the light from the 'manual_control' list, default: true |
